src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10949,99 +10946,144 @@ private ExpressionSyntax ParseSubExpressionCore(Precedence precedence)
+        /// that precedence if we have something right-associative <see cref="IsRightAssociative"/>.
+        /// </summary>
+        private ExpressionSyntax ParseExpressionContinued(ExpressionSyntax unaryOrPrimaryExpression, Precedence precedence)
         {
-            while (true)
+            var expandedExpression = unaryOrPrimaryExpression;
+
+            // Keep on expanding the left operand as long as what we see fits the precedence we're under.
+            while (tryExpandExpression(ref expandedExpression, precedence))
+                continue;
+
+            // Finally, consume a conditional expression if precedence allows it.
+
+            // https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#1115-conditional-operator:
+            //
+            // conditional_expression
+            //     : null_coalescing_expression
+            //     | null_coalescing_expression '?' expression ':' expression
+            //     ;
+            //
+            // 1. Only take the conditional part of the expression if we're at or below its precedence.
+            // 2. When parsing the branches of the expression, parse at the highest precedence again ('expression').
+            //    This allows for things like assignments/lambdas in the branches of the conditional.
+            if (this.CurrentToken.Kind == SyntaxKind.QuestionToken && precedence <= Precedence.Conditional)
+                return consumeConditionalExpression(expandedExpression);
+
+            return expandedExpression;
+
+            bool tryExpandExpression(ref ExpressionSyntax leftOperand, Precedence precedence)
             {
-                // We either have a binary or assignment operator here, or we're finished.
+                // Look for operators that can follow what we've seen so far, and which are acceptable at this
+                // precedence level.  Examples include binary operator, assignment operators, range operators `..`, as
+                // well as `switch` and `with` clauses.
                 var tk = this.CurrentToken.ContextualKind;
 
-                bool isAssignmentOperator = false;
                 SyntaxKind opKind;
 
                 // If the set of expression continuations is updated here, please review ParseStatementAttributeDeclarations
                 // to see if it may need a similar look-ahead check to determine if something is a collection expression versus
                 // an attribute.
 
-                if (IsExpectedBinaryOperator(tk))
+                // check for >>, >>=, >>> or >>>=
+                if (tk == SyntaxKind.GreaterThanToken
+                    && this.PeekToken(1) is { Kind: SyntaxKind.GreaterThanToken or SyntaxKind.GreaterThanEqualsToken } token1
+                    && NoTriviaBetween(this.CurrentToken, token1)) // check to see if they really are adjacent
```

Comment:
instead of having `>` be handled by 'IsBinaryOperator' and then undoing its logic.  we just check for the fun cases up front, and then only handle it as a real > if it doesn't match that.